facebook/OpenBIC
# Artemis: cb: Support PLDM command to notify BMC addsel

# Description
- Support PLDM command to notify BMC addsel.
- Support PMbus alert event.

# Reference GPIO table
- Artemis ACB_BIC GPIO table_20230801.xlsx

# Motivation
- Support PLDM command to notify BMC addsel.
- Support PMbus alert event.

# Test Plan
- Build code: Pass
- Enable SMB_PMBUS_ALERT_N_R GPIO pin: Pass
- Check alert event on EVT2: Pass
- Check alert event on Pre-DVT: Pass
- Check whether PMbus alert event is accidentally generated after MB on/off/sled-cycle: Pass

# Log
- Enable SMB_PMBUS_ALERT_N_R GPIO pin uart:~$ platform gpio get 0
[0  ] SMB_PMBUS_ALERT_N_R                : PP  | input (I) | 1(1)

- Check alert event on EVT2 root@bmc-oob:~# log-util all --print
2018 Mar 09 04:47:18 log-util: User cleared all logs
0    all      2018-03-09 04:48:45    pldmd            Device type: CB_P0V8_VR (0x1), board info: MAIN_SOURCE (0x0) Assert event
0    all      2018-03-09 04:48:51    pldmd            Device type: CB_P0V8_VR (0x1), board info: MAIN_SOURCE (0x0) Deassert event
0    all      2018-03-09 04:48:56    pldmd            Device type: CB_P1V25_MONITOR (0x3), board info: MAIN_SOURCE (0x0) Assert event
0    all      2018-03-09 04:49:01    pldmd            Device type: CB_P1V25_MONITOR (0x3), board info: MAIN_SOURCE (0x0) Deassert event
0    all      2018-03-09 04:49:09    pldmd            Device type: CB_P12V_ACCL2_MONITOR (0x5), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Assert event
0    all      2018-03-09 04:49:12    pldmd            Device type: CB_P12V_ACCL2_MONITOR (0x5), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Deassert event
0    all      2018-03-09 04:49:16    pldmd            Device type: CB_P12V_ACCL3_MONITOR (0x6), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Assert event
0    all      2018-03-09 04:49:21    pldmd            Device type: CB_P12V_ACCL3_MONITOR (0x6), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Deassert event

- Check alert event on Pre-DVT root@bmc-oob:~# log-util all --print
2018 Mar 09 05:01:41 log-util: User cleared all logs
0    all      2018-03-09 05:11:14    pldmd            Device type: CB_P0V8_VR (0x1), board info: MAIN_SOURCE (0x0) Assert event
0    all      2018-03-09 05:11:19    pldmd            Device type: CB_P0V8_VR (0x1), board info: MAIN_SOURCE (0x0) Deassert event
0    all      2018-03-09 05:11:35    pldmd            Device type: CB_P1V25_MONITOR (0x3), board info: MAIN_SOURCE (0x0) Assert event
0    all      2018-03-09 05:11:40    pldmd            Device type: CB_P1V25_MONITOR (0x3), board info: MAIN_SOURCE (0x0) Deassert event
0    all      2018-03-09 05:11:44    pldmd            Device type: CB_P12V_ACCL1_MONITOR (0x4), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Assert event
0    all      2018-03-09 05:11:49    pldmd            Device type: CB_P12V_ACCL1_MONITOR (0x4), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Deassert event
0    all      2018-03-09 05:11:55    pldmd            Device type: CB_P12V_ACCL2_MONITOR (0x5), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Assert event
0    all      2018-03-09 05:12:00    pldmd            Device type: CB_P12V_ACCL2_MONITOR (0x5), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Deassert event
0    all      2018-03-09 05:12:07    pldmd            Device type: CB_P12V_ACCL3_MONITOR (0x6), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Assert event
0    all      2018-03-09 05:12:11    pldmd            Device type: CB_P12V_ACCL3_MONITOR (0x6), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Deassert event
0    all      2018-03-09 05:12:17    pldmd            Device type: CB_P12V_ACCL4_MONITOR (0x7), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Assert event
0    all      2018-03-09 05:12:23    pldmd            Device type: CB_P12V_ACCL4_MONITOR (0x7), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Deassert event
0    all      2018-03-09 05:12:26    pldmd            Device type: CB_P12V_ACCL5_MONITOR (0x8), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Assert event
0    all      2018-03-09 05:12:31    pldmd            Device type: CB_P12V_ACCL5_MONITOR (0x8), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Deassert event
0    all      2018-03-09 05:12:34    pldmd            Device type: CB_P12V_ACCL6_MONITOR (0x9), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Assert event
0    all      2018-03-09 05:12:38    pldmd            Device type: CB_P12V_ACCL6_MONITOR (0x9), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Deassert event
0    all      2018-03-09 05:17:56    pldmd            Device type: CB_P12V_ACCL7_MONITOR (0xa), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Assert event
0    all      2018-03-09 05:17:58    pldmd            Device type: CB_P12V_ACCL7_MONITOR (0xa), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Deassert event
0    all      2018-03-09 05:18:24    pldmd            Device type: CB_P12V_ACCL10_MONITOR (0xd), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Assert event
0    all      2018-03-09 05:18:26    pldmd            Device type: CB_P12V_ACCL10_MONITOR (0xd), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Deassert event
0    all      2018-03-09 05:19:12    pldmd            Device type: CB_P12V_ACCL11_MONITOR (0xe), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Assert event
0    all      2018-03-09 05:19:12    pldmd            Device type: CB_P12V_ACCL11_MONITOR (0xe), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Deassert event
0    all      2018-03-09 05:19:25    pldmd            Device type: CB_P12V_ACCL12_MONITOR (0xf), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Assert event
0    all      2018-03-09 05:19:25    pldmd            Device type: CB_P12V_ACCL12_MONITOR (0xf), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Deassert event
0    all      2018-03-09 05:19:31    pldmd            Device type: CB_P12V_ACCL8_MONITOR (0xb), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Assert event
0    all      2018-03-09 05:19:33    pldmd            Device type: CB_P12V_ACCL8_MONITOR (0xb), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Deassert event
0    all      2018-03-09 05:19:37    pldmd            Device type: CB_P12V_ACCL9_MONITOR (0xc), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Assert event
0    all      2018-03-09 05:19:38    pldmd            Device type: CB_P12V_ACCL9_MONITOR (0xc), board info: MAIN_SOURCE (0x0), event type: OVER_POWER (0x1) Deassert event
0    all      2018-03-09 05:22:26    pldmd            Device type: CB_POWER_BRICK (0x2), board info: MAIN_SOURCE (0x0) Assert event
0    all      2018-03-09 05:22:53    pldmd            Device type: CB_POWER_BRICK (0x2), board info: MAIN_SOURCE (0x0) Deassert event
